src/App.tsx

```diff
@@ -1,24 +1,203 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import {
+  deleteTodo,
+  getTodos,
+  updateTodoStatus,
+  updateTodoTitle,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './Components/TodoList/TodoList';
+import { filterTodos } from './utils/helpers/filterTodos';
+import { Footer } from './Components/Footer/Footer';
+import { Header } from './Components/Header/Header';
+import { ErrorMessage } from './Components/ErrorMessage/ErrorMessage';
+import { USER_ID } from './utils/UserID';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [filteredBy, setFilteredBy] = useState('All');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [titleTodo, setTitleTodo] = useState('');
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [activeInput, setActiveInput] = useState(true);
+
+  const hasActiveTodo = todoList.some((todo) => !todo.completed);
+
+  const visibleTodos = useMemo(() => {
+    return filterTodos(todoList, filteredBy);
+  }, [todoList, filteredBy]);
+
+  const fetchTodos = async () => {
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setTodoList(todos);
+    } catch {
+      setErrorMessage('Unable to add a todo');
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  const handleChangeFilterBy = (value: string) => {
+    setFilteredBy(value);
+  };
+
+  const handleChangeStatus = useCallback(
+    async (todoID: number) => {
+      try {
+        const foundTodo = todoList.find((todo) => todo.id === todoID);
+
+        setLoadingTodos((prevIDs) => [...prevIDs, todoID]);
+        await updateTodoStatus(todoID, !foundTodo?.completed);
+
+        const newTodos = todoList.map((todo) => {
+          if (foundTodo?.id === todo.id) {
+            foundTodo.completed = !foundTodo.completed;
+
+            return foundTodo;
+          }
+
+          return todo;
+        });
+
+        setTodoList(newTodos);
+      } catch {
+        setErrorMessage('Unable to update a todo');
+      } finally {
+        setLoadingTodos([]);
+      }
+    },
+    [todoList],
+  );
+
+  const handleUpdateTitleTodo = useCallback(
+    async (todoID: number, todoNewTitle: string) => {
+      if (!todoNewTitle) {
+        try {
+          setLoadingTodos([todoID]);
+          const newTodos = todoList.filter((todo) => todo.id !== todoID);
+
+          await deleteTodo(`/todos/${todoID}`);
```

Comment:
Create functions for making request, endpoints can be changed and it is better to maintain it in one place